composer.json

```diff
@@ -6,6 +6,7 @@
 	],
 	"require": {
 		"php": "~7.1",
+		"ext-filter": "*",
```

Comment:
I think this should be in require-dev, as it is not required for analysis.

Reply:
Nope, it's required. The constants are used in the type extension rule in the `src` directory. Remember that this doesn't mean your application will require it in production.

Reply:
> as it is not required for analysis

To elaborate on this, it is required since I use the `FILTER_` constants directly. I could probably do an AST string comparison.

Reply:
And what if I don't use filter_var function in my application? Then phpstan will not use this extension. Is it required then?

Reply:
Yes. Remember that `ext-filter` is enabled by default. We also don't check if you have `pcre` enabled, although you could technically disable it. Most major libraries depend on it. The only reason I added it is because the build fails without it (composer-require-checker).

Reply:
I got rid of the dependency.